### ISSUE TITLE
Add default JSON path

### ISSUE THREAD

**Issue by gabe-microsoft** [mpkocher/pydantic-cli#14]
First, I love this package and it makes creating CLIs with Pydantic really nice! 

It would be nice if there were an option in the model Config to specify a default JSON path (say CLI_JSON_DEFAULT_PATH). If this default path is defined, then the CLI_JSON_KEY command line argument would be optional and settings would be loaded from the default JSON file if it exists (no error if it does not). If the CLI_JSON_KEY argument is provided on the command line, then it will override the default value.

This seems pretty straightforward to add and would allow a common pattern that my team uses in which the default JSON config file is SCRIPT_NAME_config.json. This way the script can be run without arguments as long as the default JSON file specifies a complete model.

**Comment by mpkocher** [comment]
Hi @gabe-microsoft, thanks for the feedback. This seems reasonable and straight-forward to add. 

However, I believe that if the file doesn't exist, there needs to be an error raised to avoid unexpected behavior and to overall be consistent. As long as the file exists and is a valid empty json file, this will satisfy the interface. Does these seem reasonable? 

As a workaround (and what I currently do for similar cases) is to create a bash alias or function with the default location of the preset JSON file. 

For example,

```bash
alias analyze-twitter-default="$HOME/bin/analyze-twitter --json-config=$HOME/configs/analyze-twitter-python-retrain.json"
```

Something tangentially related is setting values of Pydantic models or the default value `--json-config`  from ENV vars. This **doesn't** really work as expected (yet) in pydantic-cli.

**Comment by gabe-microsoft** [comment]
Sorry for the slow response. I think it would be best not to error if the JSON file does not exist. Of course, there will likely be errors if not all required variables are defined, but I would prefer all errors coming from pydantic based on the validity of the settings, and I don't think the lack of a JSON config file should be an error. I think of the JSON config file as a source of environment-specific settings (exactly like grabbing settings from ENV vars, which like you say doesn't currently work).

Regarding loading settings from ENV vars, I think pydantic-cli is very close to supporting this. This is based on my memory from some testing from 3 weeks ago, so forgive me is some of this is incorrect. I think there are two issues preventing the use of BaseSettings classes rather than BaseModel classes with pydantic-cli:
1. Validation within argparse. Since a valid BaseSettings object may only be valid once ENV vars are loaded (i.e., incomplete based on command line args and possibly JSON file), no validation should be done within argparse. Basically, if all fields/arguments are considered not required as far as argparse is concerned, validation will only be done by pydantic and this will allow the use of BaseSettings.
2. Pydantic-cli seems to instantiate the pydantic BaseModel object with extra arguments (cls, func, and config_file). By default these are ignored with a BaseModel because BaseModel.Config.extra is 'ignore' by default. However, the default for BaseSettings.Config.extra seems to be 'forbid', thus causing a validation error when instantiated using pydantic-cli. The user can certainly override the Config.extra field to 'allow' or 'ignore' to allow the use of BaseSettings in pydantic-cli, but is it possible to change the way you instantiate the object so that you don't pass in the extra args?

**Comment by mpkocher** [comment]
I believe this should be resolved in version 3.1.0

Three new configuration values are provided.

1. CLI_JSON_CONFIG_ENV_VAR: str = "PCLI_JSON_CONFIG" # Set the default ENV var for defining the JSON config path
1. CLI_JSON_CONFIG_PATH: T.Optional[str] = None # Set the default Path for JSON config file
1. CLI_JSON_VALIDATE_PATH: bool = True    # If a default path is provided or provided from the commandline ignore the file. This should perhaps be used judiciously and enabled in only very specific cases.

An issue is also fixed where the `Config.extra` wouldn't work as expected due to pollution of the Pydantic data model (via argparse namespace). 

With regards to `BaseSettings`, I investigated this feature and I decided that it didn't really make sense to support. The feature overlapped and duplicated  too much with the JSON loading core feature of pydantic-cli.  

That's my current read of the room, however this can be reevaluated in the future. 

Thanks again for the feedback.
